Sofa/framework/Type/test/MatTypes_test.cpp

```diff
@@ -376,3 +376,41 @@ TEST(MatTypesTest, conversionToReal)
     const SReal p = id.toReal();
     EXPECT_EQ(p, 1_sreal);
 }
+
+TEST(MatTypesTest, fromPtr)
+{
+    const float arrayMat3f[9]{ 1.0f, 2.0f, 3.0f, 2.0f, 4.0f, 6.0f, 3.0f, 6.0f, 9.0f };
+    sofa::type::Mat<3, 3, float> mat3f(arrayMat3f);
+
+    bool comp = true;
+    for (std::size_t i = 0; i < 3; i++)
+        for (std::size_t j = 0; j < 3; j++)
+            comp = (arrayMat3f[i * 3 + j] == mat3f[i][j]) && comp;
+    
+    EXPECT_TRUE(comp);
+
+    sofa::type::Mat<3, 3, double> mat3d(arrayMat3f);
+
+    bool comp2 = true;
+    constexpr double epsilon = 0.00001;
+    for (std::size_t i = 0; i < 3; i++)
+        for (std::size_t j = 0; j < 3; j++)
+            comp = (std::fabs(arrayMat3f[i * 3 + j] - mat3d[i][j]) < epsilon) && comp2;
```

Comment:
```suggestion
            comp2 = (std::fabs(arrayMat3f[i * 3 + j] - mat3d[i][j]) < epsilon) && comp2;
```
But I recommend 2 different tests. Or at least, two different scopes